File: src/tools/lizard/eval-repos/synthetic/go/edge_cases/unicode.go
// Package edge_cases provides Unicode content tests.
package edge_cases

// Unicode in strings
var greeting = "Hello, 世界! 🌍"
var emojiMath = "1️⃣ + 2️⃣ = 3️⃣"

// Translations with various Unicode
var translations = map[string]string{
	"hello":   "你好",
	"world":   "мир",
	"welcome": "مرحبا",
	"goodbye": "さようなら",
	"thanks":  "धन्यवाद",
}

// GetTranslation returns a translation.
func GetTranslation(key string) string {
	if val, ok := translations[key]; ok {
		return val
	}
	return ""
}

// FormatGreeting creates a multilingual greeting.
func FormatGreeting(name string) string {
	return "Привет, " + name + "! 👋 Welcome to 日本!"
}

// GetStatus returns a status with emoji.
func GetStatus(success bool) string {
	if success {
		return "✅ Success"
	}
	return "❌ Failed"
}

File: src/tools/lizard/eval-repos/synthetic/javascript/edge_cases/unicode.js
/**
 * Unicode content test file.
 */

// Unicode in strings
const greeting = 'Hello, 世界! 🌍';
const emojiMath = '1️⃣ + 2️⃣ = 3️⃣';

// Various Unicode characters
const translations = {
  hello: '你好',
  world: 'мир',
  welcome: 'مرحبا',
  goodbye: 'さようなら',
  thanks: 'धन्यवाद',
};

function getTranslation(key) {
  return translations[key] ?? null;
}

// Unicode in template literals
function formatGreeting(name) {
  return `Привет, ${name}! 👋 Welcome to 日本!`;
}

// Emoji in function
function getStatus(success) {
  return success ? '✅ Success' : '❌ Failed';
}

// Unicode regex
const emojiPattern = /[\u{1F300}-\u{1F9FF}]/gu;

function countEmojis(text) {
  const matches = text.match(emojiPattern);
  return matches ? matches.length : 0;
}

module.exports = {
  greeting,
  emojiMath,
  translations,
  getTranslation,
  formatGreeting,
  getStatus,
  countEmojis,
};
